Sources/FoundationInternationalization/ICU/ICU+StringConverter.swift

```diff
@@ -0,0 +1,202 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(FoundationEssentials)
+import FoundationEssentials
+#endif
+internal import _FoundationICU
+
+private extension String.Encoding {
+    var _icuConverterName: String? {
```

Comment:
Should we make sure this returns `nil` for any encoding which we never want to use ICU for? `UTF8`, `ASCII`, `UTF16/32`, etc.

Reply:
Certainly.
I'll let them return `nil` here or in `ICU.StringConverter`'s `init`.